### ISSUE TITLE
Tests backward compatibility

### ISSUE THREAD

**Issue by h2zh** [PelicanPlatform/pelican#1785]
- Run latest Pelican binary in the container built for Pelican v7.8 release and run tests
- Run v7.8 release in the latest container and run tests

**Comment by h2zh** [comment]
tl;dr: No, they don't work. If the tag mismatched binary and image are put together, they crash badly.
<google-sheets-html-origin><style type="text/css"></style>
  |   | pelican binary version
-- | -- | --
  |   | main (latest) | v7.8 release (v7.8.9)
pelican container image version | v7.11.7 (latest) | \ | all tests passed; goreleaser build succeeded; Pelican running failed*
v7.8 release (v7.8.9) | unit test (TestHttpOriginConfig) failed** | \

<google-sheets-html-origin><style type="text/css"></style>
note: latest = as of Dec 3; all the tests happen in the same container (which means no cross-container test)
--
*error message: http: TLS handshake error from 172.17.0.2:45812: remote error: tls: bad certificate
**error message: Xrootd initialization failed. So there's no need for goreleaser and Pelican running tests
These images doesn't contain many necessary packages, including Go, Goreleaser, make, Node.js, npm

**Comment by h2zh** [comment]
Running a container built from the images hosted on Harbor could be challenging. Here is my solution for macOS users:

**Comment by bbockelm** [comment]
> error message: http: TLS handshake error from 172.17.0.2:45812: remote error: tls: bad certificate

That suggests a simple TLS configuration issue, not anything systematic.  What's causing it?

**Comment by h2zh** [comment]
> > error message: http: TLS handshake error from 172.17.0.2:45812: remote error: tls: bad certificate
> 
> That suggests a simple TLS configuration issue, not anything systematic. What's causing it?

It fails at the beginning of registry initialization. I double checked that TLSSkipVerify is set to true. I also tried to delete /dist and /run/pelican and build again, but it doesn't work. Here is the full log

> [root@87f35ac5cf5f pelican_linux_arm64_v8.0]# ./pelican registry serve -p 8446
> Version: 7.8.10-next
> Build Date: 2024-12-04T15:56:36Z
> Build Commit: b93a59097d088aee28b84279efc2d9e7ba555e78
> Built By: goreleaser
> 2024/12/04 16:05:59 OK   20240212192712_create_db_tables.sql (2.29ms)
> 2024/12/04 16:05:59 OK   20240212194951_change_admin_metadata_default.sql (3.13ms)
> 2024/12/04 16:05:59 goose: successfully migrated database to version: 20240212194951
> 2024/12/04 16:05:59 http: TLS handshake error from 172.17.0.2:39092: remote error: tls: bad certificate
> 2024/12/04 16:05:59 http: TLS handshake error from 172.17.0.2:39100: remote error: tls: bad certificate
> 2024/12/04 16:05:59 http: TLS handshake error from 172.17.0.2:39116: remote error: tls: bad certificate
> ...
> ^CERROR[2024-12-04T16:06:03Z] Web engine check failed:  context canceled   
> Error: context canceled
> ERROR[2024-12-04T16:06:03Z] Fatal error occurred at the start of the program. Cleanup started: context canceled 
> Pelican is safely exited

**Comment by h2zh** [comment]
Thanks for your pro-tip @turetske! I just updated the result above. Now running the v7.8.9 code/binary in latest (v7.11.7) container is all good, but the error persists in the other way. I also add a "controlled group" as the above shown.

Troubleshooting note: When I run multiple pelican containers, the `https://localhost:8446/view/registry/` website could not be reached maybe due to some proxy conflicts. To fix it, just stop the container I'm not currently using.

**Comment by turetske** [comment]
@h2zh Most likely because @bbockelm added some configuration variables that are in the newer cache.config and not recognized with older xrootd.

**Comment by turetske** [comment]
More specifically, the problematic lines are probably line 30 and lines 61-64 in `xrootd/resource/xrootd-cache.cfg`

**Comment by bbockelm** [comment]
Why do we care about older xrootd with newer Pelican?

The project is to run an older Pelican in the CI pipeline - which would imply a new xrootd, no?

**Comment by turetske** [comment]
@bbockelm Yes, but I think you mentioned you wanted to test both ways, so that was done. Possibly as misunderstanding on our part. I agree with you.
